Update app lifecycle troubleshooting doc in 2.4

xiangjingli opened the pull request:
https://github.com/open-cluster-management/backlog/issues/16822

xiangjingli committed:
Update app lifecycle troubleshooting doc in 2.4

dockerymick commented on troubleshooting/trouble_local_cluster.adoc:
```diff
@@ -1,17 +1,17 @@
 = Troubleshooting local cluster not selected with placement rule
 
-The managed clusters are selected with a placement rule, but the `local-cluster` (hub cluster that is also managed) is not selected. The placement rule user is not granted to permission to create deployable resources in the `local-cluster` namespace.
+The managed clusters are selected with a placement rule, but the `local-cluster` (hub cluster that is also managed) is not selected. The placement rule user is not granted permission to get the managedcluster resources in the `local-cluster` namespace.
```
@cadawson whenever you get to this, I recommend the following change `managedcluster` resources

I can also update the file if you need some help w/ PRs

swopebe replied:
I will go ahead and take this one.

swopebe committed:
open-cluster-management/backlog#16822 changes during review.

@xiangjingli I made some necessary changes--please let me know if there are any issues.

swopebe reviewed (approved):
approve after edits

xiangjingli commented:
@swopebe  Thanks! The new updates looks good!
